### DISCUSSION TITLE
Dynamically create/delete PDU Session

### DISCUSSION THREAD

**Discussion by brianchennn** [HewlettPackard/PacketRusher#116] [Q&A]
@linouxis9 
I wonder if it's possible to create another PDU Session dynamically for UE after executing `sudo ./packetrusher ue`. I want to test if free5gc can handle NGAP/NAS message correctly after AMF planned removal.

**Comment by linouxis9** [comment]
Hi @brianchennn,
Sadly, it's not yet supported, it's in our TODO list.
In the meanwhile, you can tweak the scenario here to do what you want to achieve: https://github.com/HewlettPackard/PacketRusher/blob/46f1706201bc6a27b23ce924ea865093351cc5da/internal/common/tools/tools.go#L155
ex: add in this file something like:
```go
go func() {
  for {
    time.Sleep(time.Second*5)
    ueRx <- procedures.UeTesterMessage{Type: procedures.DestroyPDUSession, Param: 1}
    time.Sleep(time.Second*5)
    ueRx <- procedures.UeTesterMessage{Type: procedures.NewPDUSession, Param: 1}
  }()
}
```
